nservicebus/upgrades/7to8/message-contracts.md

```diff
@@ -0,0 +1,47 @@
+---
+title: Upgrading message contracts from Version 7 to 8
+summary: Instructions on how to upgrade shared message contracts from NServiceBus version 7 to version 8.
+reviewed: 2022-02-02
+component: Core
+isUpgradeGuide: true
+upgradeGuideCoreVersions:
+ - 7
+ - 8
+---
+
+Since NServiceBus version 8 no longer supports .NET Standard, message contract assemblies refrencing NServiceBus might need to be updated. This guidance outlines possible approaches.
+
+Note: When using [unobtrusive mode](TODO), the contracts assembly doesn't require a reference to NServiceBus and therefore is not affected. Unobtrusive message contracts can continue to target .NET Standard.
+
+## Change to specifc target Platform
+
+If all endpoints are targeting the same platform (e.g. .NET Core 3.1), the message contracts assembly can be changed to be aligned with all other endpoint projects.
+
+## Multi-targeting
+
+If endpoints sharing a contracts assembly are targeting different platforms and frameworks (e.g. .NET Framework 4.8 and .NET Core 3.1), the target assembly can be changed to use [multi-targeting](https://docs.microsoft.com/en-us/dotnet/standard/library-guidance/cross-platform-targeting#multi-targeting) by replacing the `TargetFramwork` element with the `TargetFrameworks` element in the `.csproj` settings:
+
+```xml
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <!-- Typically, it is sufficient to only list the lowest version needed for each platform -->
+    <TargetFrameworks>net48;netcoreapp3.1</TargetFrameworks>
+  </PropertyGroup>
+</Project>
+```
+
+## Migration
```

Comment:
This seems like more general advice that could apply outside of the 7 -> 8 migration, yesno?

Reply:
yes probably something that should be part of the "evolving message contracts" page and we can link to that. I didn't want to touch both files in the same PR though. maybe we need to start with the other one first?

Reply:
I don't see the problem of including that file in this PR. However, I agree that it makes sense to move the section there.